<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Adam Mattsson" />

<meta name="date" content="2025-10-16" />

<title>Data Retrieval and Utility Workflows</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Data Retrieval and Utility Workflows</h1>
<h4 class="author">Adam Mattsson</h4>
<h4 class="date">2025-10-16</h4>


<div id="TOC">
<ul>
<li><a href="#introduction" id="toc-introduction"><span class="toc-section-number">1</span> Introduction</a>
<ul>
<li><a href="#whats-covered" id="toc-whats-covered"><span class="toc-section-number">1.1</span> What’s Covered</a></li>
<li><a href="#extracting-prediction-subsets-with-get_prediction_subset" id="toc-extracting-prediction-subsets-with-get_prediction_subset"><span class="toc-section-number">1.2</span> 1. Extracting Prediction Subsets
with <code>get_prediction_subset</code></a>
<ul>
<li><a href="#example-subsetting-by-sample-ids" id="toc-example-subsetting-by-sample-ids"><span class="toc-section-number">1.2.1</span> Example: Subsetting by Sample
IDs</a></li>
<li><a href="#example-subsetting-by-metadata" id="toc-example-subsetting-by-metadata"><span class="toc-section-number">1.2.2</span> Example: Subsetting by
Metadata</a></li>
</ul></li>
<li><a href="#ordering-samples-by-proliferation-score-with-get_sample_order" id="toc-ordering-samples-by-proliferation-score-with-get_sample_order"><span class="toc-section-number">1.3</span> 2. Ordering Samples by
Proliferation Score with <code>get_sample_order</code></a>
<ul>
<li><a href="#example-ordering-by-lateearly-ratio" id="toc-example-ordering-by-lateearly-ratio"><span class="toc-section-number">1.3.1</span> Example: Ordering by Late/Early
Ratio</a></li>
</ul></li>
<li><a href="#summarizing-subtype-metrics-with-get_subtype_metrics" id="toc-summarizing-subtype-metrics-with-get_subtype_metrics"><span class="toc-section-number">1.4</span> 3. Summarizing Subtype Metrics
with <code>get_subtype_metrics</code></a>
<ul>
<li><a href="#example-counting-events-by-subtype" id="toc-example-counting-events-by-subtype"><span class="toc-section-number">1.4.1</span> Example: Counting Events by
Subtype</a></li>
<li><a href="#example-summarizing-gender-distribution" id="toc-example-summarizing-gender-distribution"><span class="toc-section-number">1.4.2</span> Example: Summarizing Gender
Distribution</a></li>
</ul></li>
</ul></li>
</ul>
</div>

<div id="introduction" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Introduction</h1>
<p>This vignette demonstrates how to efficiently retrieve, filter, and
summarize classification results using LundTaxR’s utility functions.
After classifying samples, you often need to extract specific subsets,
order samples for visualization, or compute summary metrics for
downstream analysis. LundTaxR provides a suite of helper functions to
streamline these common tasks.</p>
<div id="whats-covered" class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> What’s Covered</h2>
<p>In this vignette, you’ll learn how to:</p>
<ul>
<li><strong>Extract subsets of predictions</strong> for specific sample
groups using <code>get_prediction_subset</code></li>
<li><strong>Order samples</strong> based on subtype, score, or other
criteria with <code>get_sample_order</code></li>
<li><strong>Compute and interpret subtype metrics</strong> using
<code>get_subtype_metrics</code> for summary statistics and quality
control</li>
</ul>
<p>Each section includes practical code examples that you can adapt to
your own data and workflows.</p>
<hr />
</div>
<div id="extracting-prediction-subsets-with-get_prediction_subset" class="section level2" number="1.2">
<h2><span class="header-section-number">1.2</span> 1. Extracting
Prediction Subsets with <code>get_prediction_subset</code></h2>
<p>After classifying your samples, you may want to focus your analysis
on a specific subset of samples—such as those belonging to a particular
group or cohort. The <code>get_prediction_subset</code> function is
designed to help you do just that: it extracts only the relevant samples
from all components of your prediction results, whether those components
are data frames or named vectors.</p>
<p>You can specify the samples to keep either by providing a vector of
sample IDs, or by passing a metadata data frame that contains the
desired samples. The function will return a new predictions list, but
restricted to just the samples you specified.</p>
<div id="example-subsetting-by-sample-ids" class="section level3" number="1.2.1">
<h3><span class="header-section-number">1.2.1</span> Example: Subsetting
by Sample IDs</h3>
<p>Suppose you want to examine only a handful of samples from your
classification results:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="co"># Run the classifier (if not already done)</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sjodahl_2017&quot;</span>)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sjodahl_2017_meta&quot;</span>)</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a>results <span class="ot">&lt;-</span> <span class="fu">classify_samples</span>(</span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a>  <span class="at">this_data =</span> sjodahl_2017,</span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a>  <span class="at">include_data =</span> <span class="cn">TRUE</span>,</span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a>  <span class="at">verbose =</span> <span class="cn">FALSE</span></span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a>)</span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a></span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a><span class="co"># Select a few sample IDs</span></span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a>selected_samples <span class="ot">&lt;-</span> <span class="fu">head</span>(sjodahl_2017_meta<span class="sc">$</span>sample_id)</span>
<span id="cb1-12"><a href="#cb1-12" tabindex="-1"></a></span>
<span id="cb1-13"><a href="#cb1-13" tabindex="-1"></a><span class="co"># Subset the results to just these samples</span></span>
<span id="cb1-14"><a href="#cb1-14" tabindex="-1"></a>subset_results <span class="ot">&lt;-</span> <span class="fu">get_prediction_subset</span>(</span>
<span id="cb1-15"><a href="#cb1-15" tabindex="-1"></a>  <span class="at">these_predictions =</span> results,</span>
<span id="cb1-16"><a href="#cb1-16" tabindex="-1"></a>  <span class="at">these_sample_ids =</span> selected_samples</span>
<span id="cb1-17"><a href="#cb1-17" tabindex="-1"></a>)</span>
<span id="cb1-18"><a href="#cb1-18" tabindex="-1"></a></span>
<span id="cb1-19"><a href="#cb1-19" tabindex="-1"></a><span class="co"># View the subsetted data</span></span>
<span id="cb1-20"><a href="#cb1-20" tabindex="-1"></a><span class="fu">head</span>(subset_results<span class="sc">$</span>data)</span>
<span id="cb1-21"><a href="#cb1-21" tabindex="-1"></a><span class="co">#&gt;           1.CEL    2.CEL     3.CEL    4.CEL    5.CEL    6.CEL</span></span>
<span id="cb1-22"><a href="#cb1-22" tabindex="-1"></a><span class="co">#&gt; A1CF   4.248274 4.153923  4.186689 4.877813 4.106269 5.278318</span></span>
<span id="cb1-23"><a href="#cb1-23" tabindex="-1"></a><span class="co">#&gt; A2M    8.348800 7.969504 10.755738 8.467495 8.979222 9.520587</span></span>
<span id="cb1-24"><a href="#cb1-24" tabindex="-1"></a><span class="co">#&gt; A2ML1  6.694896 4.442973  8.871067 4.392010 7.685045 4.493648</span></span>
<span id="cb1-25"><a href="#cb1-25" tabindex="-1"></a><span class="co">#&gt; A4GALT 7.255671 6.337239  6.925670 6.239185 6.032115 6.539985</span></span>
<span id="cb1-26"><a href="#cb1-26" tabindex="-1"></a><span class="co">#&gt; A4GNT  3.740418 3.830598  3.999331 3.813662 3.718752 3.567144</span></span>
<span id="cb1-27"><a href="#cb1-27" tabindex="-1"></a><span class="co">#&gt; AAAS   6.837006 7.525639  6.778506 6.658134 7.354926 6.611625</span></span>
<span id="cb1-28"><a href="#cb1-28" tabindex="-1"></a><span class="fu">head</span>(subset_results<span class="sc">$</span>subtype_scores)</span>
<span id="cb1-29"><a href="#cb1-29" tabindex="-1"></a><span class="co">#&gt;          Uro   UroA   UroB   UroC     GU   BaSq    Mes   ScNE prediction_delta</span></span>
<span id="cb1-30"><a href="#cb1-30" tabindex="-1"></a><span class="co">#&gt; 1.CEL 0.9924 0.9350 0.0048 0.0602 0.0072 0.0004 0.0000 0.0000           0.0574</span></span>
<span id="cb1-31"><a href="#cb1-31" tabindex="-1"></a><span class="co">#&gt; 2.CEL 0.0016     NA     NA     NA 0.0052 0.0054 0.9804 0.0074           0.9730</span></span>
<span id="cb1-32"><a href="#cb1-32" tabindex="-1"></a><span class="co">#&gt; 3.CEL 0.0682     NA     NA     NA 0.7754 0.0834 0.0390 0.0340           0.6920</span></span>
<span id="cb1-33"><a href="#cb1-33" tabindex="-1"></a><span class="co">#&gt; 4.CEL 0.9930 0.9334 0.0028 0.0638 0.0066 0.0004 0.0000 0.0000           0.0596</span></span>
<span id="cb1-34"><a href="#cb1-34" tabindex="-1"></a><span class="co">#&gt; 5.CEL 0.9808 0.0058 0.9878 0.0064 0.0076 0.0090 0.0006 0.0020           0.0070</span></span>
<span id="cb1-35"><a href="#cb1-35" tabindex="-1"></a><span class="co">#&gt; 6.CEL 0.0082     NA     NA     NA 0.9890 0.0002 0.0004 0.0022           0.9808</span></span>
<span id="cb1-36"><a href="#cb1-36" tabindex="-1"></a><span class="fu">head</span>(subset_results<span class="sc">$</span>predictions_5classes)</span>
<span id="cb1-37"><a href="#cb1-37" tabindex="-1"></a><span class="co">#&gt; 1.CEL 2.CEL 3.CEL 4.CEL 5.CEL 6.CEL </span></span>
<span id="cb1-38"><a href="#cb1-38" tabindex="-1"></a><span class="co">#&gt; &quot;Uro&quot; &quot;Mes&quot;  &quot;GU&quot; &quot;Uro&quot; &quot;Uro&quot;  &quot;GU&quot;</span></span></code></pre></div>
</div>
<div id="example-subsetting-by-metadata" class="section level3" number="1.2.2">
<h3><span class="header-section-number">1.2.2</span> Example: Subsetting
by Metadata</h3>
<p>Alternatively, you can subset your results using a metadata data
frame. This is useful if you have filtered your metadata for a specific
group (e.g., only tumor samples, or only a certain clinical stage):</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="co"># Take a subset of the metadata</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>meta_subset <span class="ot">&lt;-</span> <span class="fu">head</span>(sjodahl_2017_meta)</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="co"># Subset the results using the metadata</span></span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a>subset_results_meta <span class="ot">&lt;-</span> <span class="fu">get_prediction_subset</span>(</span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a>  <span class="at">these_predictions =</span> results,</span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a>  <span class="at">these_samples_metadata =</span> meta_subset,</span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a>  <span class="at">samples_rownames =</span> <span class="cn">FALSE</span></span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a>)</span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a></span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a><span class="co"># View the subsetted data</span></span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a><span class="fu">head</span>(subset_results_meta<span class="sc">$</span>data)</span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a><span class="co">#&gt;           1.CEL    2.CEL     3.CEL    4.CEL    5.CEL    6.CEL</span></span>
<span id="cb2-14"><a href="#cb2-14" tabindex="-1"></a><span class="co">#&gt; A1CF   4.248274 4.153923  4.186689 4.877813 4.106269 5.278318</span></span>
<span id="cb2-15"><a href="#cb2-15" tabindex="-1"></a><span class="co">#&gt; A2M    8.348800 7.969504 10.755738 8.467495 8.979222 9.520587</span></span>
<span id="cb2-16"><a href="#cb2-16" tabindex="-1"></a><span class="co">#&gt; A2ML1  6.694896 4.442973  8.871067 4.392010 7.685045 4.493648</span></span>
<span id="cb2-17"><a href="#cb2-17" tabindex="-1"></a><span class="co">#&gt; A4GALT 7.255671 6.337239  6.925670 6.239185 6.032115 6.539985</span></span>
<span id="cb2-18"><a href="#cb2-18" tabindex="-1"></a><span class="co">#&gt; A4GNT  3.740418 3.830598  3.999331 3.813662 3.718752 3.567144</span></span>
<span id="cb2-19"><a href="#cb2-19" tabindex="-1"></a><span class="co">#&gt; AAAS   6.837006 7.525639  6.778506 6.658134 7.354926 6.611625</span></span></code></pre></div>
</div>
</div>
<div id="ordering-samples-by-proliferation-score-with-get_sample_order" class="section level2" number="1.3">
<h2><span class="header-section-number">1.3</span> 2. Ordering Samples
by Proliferation Score with <code>get_sample_order</code></h2>
<p>When visualizing or analyzing your classification results, it’s often
helpful to arrange samples in a biologically meaningful order. The
<code>get_sample_order</code> function provides a convenient way to sort
your samples based on their proliferation score, specifically the
late/early gene expression ratio. Note, this is the implemented sample
order that is shown in the <code>plot_classification_heatmap</code>.</p>
<p>This function is especially useful for downstream plotting, as it
returns a vector of sample names ordered by their proliferation status.
You can use this order to consistently arrange samples in heatmaps or
other visualizations.</p>
<div id="example-ordering-by-lateearly-ratio" class="section level3" number="1.3.1">
<h3><span class="header-section-number">1.3.1</span> Example: Ordering
by Late/Early Ratio</h3>
<p>To use <code>get_sample_order</code>, make sure you ran
<code>classify_samples</code> with <code>include_data = TRUE</code> so
the expression data is available:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="co"># Get the sample order based on late/early ratio (default)</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>ordered_samples <span class="ot">&lt;-</span> <span class="fu">get_sample_order</span>(<span class="at">expr_data =</span> results<span class="sc">$</span>data)</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="co">#&gt;   [1] 158 194 142 171  43  76 235  83 160  31  68 237 232 187  22 130 259 163</span></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a><span class="co">#&gt;  [19] 134 245  55 168 131 234  99  47 161  61 137   7  85 236 199  28 264 175</span></span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a><span class="co">#&gt;  [37] 254 202 266  13  94 184   3 253 231  26  54 164  95 103 178 179 221 223</span></span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a><span class="co">#&gt;  [55]  73 224 132 113  46  49 102 166 241 112  97 100  41 167 107 249 239  15</span></span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a><span class="co">#&gt;  [73] 220  91   9   4 114 242  64  36 119  62 233  56 243  29  98 191 246 225</span></span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a><span class="co">#&gt;  [91] 247 205  63 222 240 206 227 156  14  79  21 252  58 190  65   1 248 145</span></span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a><span class="co">#&gt; [109] 117 125 151 200 251 101 155 172 215  51 212 165  78  69  77 162  75  20</span></span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a><span class="co">#&gt; [127] 218  81 229  24 116 244 126 170 108 105 123  60 141 136 256 133  80 111</span></span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a><span class="co">#&gt; [145] 138  23  19  48 208  93 159 188  72 216  12 217 192  25 209 147 182   5</span></span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a><span class="co">#&gt; [163] 181 122 110  18 230  92  27 124  86 135 257 228 152  66  71 183 118  10</span></span>
<span id="cb3-13"><a href="#cb3-13" tabindex="-1"></a><span class="co">#&gt; [181] 204  44  50 121  57  37 250 186  40 197 153 115 219 203  89  30 177  90</span></span>
<span id="cb3-14"><a href="#cb3-14" tabindex="-1"></a><span class="co">#&gt; [199]  96 139 143  42 201 144  11 106  70 176 150 260 154   6 169 238 185 193</span></span>
<span id="cb3-15"><a href="#cb3-15" tabindex="-1"></a><span class="co">#&gt; [217] 207 180  87  33 173  16 149 104 262 195  35 127 258 109 129  59 261  32</span></span>
<span id="cb3-16"><a href="#cb3-16" tabindex="-1"></a><span class="co">#&gt; [235]  52 211 213 148 263  82  38 120   2  45  53  84  74 196 140 174 267 146</span></span>
<span id="cb3-17"><a href="#cb3-17" tabindex="-1"></a><span class="co">#&gt; [253]  39  67 265  88  17 210 255 214 189 226 128   8  34 157 198</span></span>
<span id="cb3-18"><a href="#cb3-18" tabindex="-1"></a><span class="fu">head</span>(ordered_samples)</span>
<span id="cb3-19"><a href="#cb3-19" tabindex="-1"></a><span class="co">#&gt; [1] 158 194 142 171  43  76</span></span>
<span id="cb3-20"><a href="#cb3-20" tabindex="-1"></a></span>
<span id="cb3-21"><a href="#cb3-21" tabindex="-1"></a><span class="co"># If you want the actual late/early ratio values instead of the order:</span></span>
<span id="cb3-22"><a href="#cb3-22" tabindex="-1"></a>late_early_ratios <span class="ot">&lt;-</span> <span class="fu">get_sample_order</span>(<span class="at">expr_data =</span> results<span class="sc">$</span>data, <span class="at">return_this =</span> <span class="st">&quot;late_early&quot;</span>)</span>
<span id="cb3-23"><a href="#cb3-23" tabindex="-1"></a><span class="co">#&gt;   [1] 158 194 142 171  43  76 235  83 160  31  68 237 232 187  22 130 259 163</span></span>
<span id="cb3-24"><a href="#cb3-24" tabindex="-1"></a><span class="co">#&gt;  [19] 134 245  55 168 131 234  99  47 161  61 137   7  85 236 199  28 264 175</span></span>
<span id="cb3-25"><a href="#cb3-25" tabindex="-1"></a><span class="co">#&gt;  [37] 254 202 266  13  94 184   3 253 231  26  54 164  95 103 178 179 221 223</span></span>
<span id="cb3-26"><a href="#cb3-26" tabindex="-1"></a><span class="co">#&gt;  [55]  73 224 132 113  46  49 102 166 241 112  97 100  41 167 107 249 239  15</span></span>
<span id="cb3-27"><a href="#cb3-27" tabindex="-1"></a><span class="co">#&gt;  [73] 220  91   9   4 114 242  64  36 119  62 233  56 243  29  98 191 246 225</span></span>
<span id="cb3-28"><a href="#cb3-28" tabindex="-1"></a><span class="co">#&gt;  [91] 247 205  63 222 240 206 227 156  14  79  21 252  58 190  65   1 248 145</span></span>
<span id="cb3-29"><a href="#cb3-29" tabindex="-1"></a><span class="co">#&gt; [109] 117 125 151 200 251 101 155 172 215  51 212 165  78  69  77 162  75  20</span></span>
<span id="cb3-30"><a href="#cb3-30" tabindex="-1"></a><span class="co">#&gt; [127] 218  81 229  24 116 244 126 170 108 105 123  60 141 136 256 133  80 111</span></span>
<span id="cb3-31"><a href="#cb3-31" tabindex="-1"></a><span class="co">#&gt; [145] 138  23  19  48 208  93 159 188  72 216  12 217 192  25 209 147 182   5</span></span>
<span id="cb3-32"><a href="#cb3-32" tabindex="-1"></a><span class="co">#&gt; [163] 181 122 110  18 230  92  27 124  86 135 257 228 152  66  71 183 118  10</span></span>
<span id="cb3-33"><a href="#cb3-33" tabindex="-1"></a><span class="co">#&gt; [181] 204  44  50 121  57  37 250 186  40 197 153 115 219 203  89  30 177  90</span></span>
<span id="cb3-34"><a href="#cb3-34" tabindex="-1"></a><span class="co">#&gt; [199]  96 139 143  42 201 144  11 106  70 176 150 260 154   6 169 238 185 193</span></span>
<span id="cb3-35"><a href="#cb3-35" tabindex="-1"></a><span class="co">#&gt; [217] 207 180  87  33 173  16 149 104 262 195  35 127 258 109 129  59 261  32</span></span>
<span id="cb3-36"><a href="#cb3-36" tabindex="-1"></a><span class="co">#&gt; [235]  52 211 213 148 263  82  38 120   2  45  53  84  74 196 140 174 267 146</span></span>
<span id="cb3-37"><a href="#cb3-37" tabindex="-1"></a><span class="co">#&gt; [253]  39  67 265  88  17 210 255 214 189 226 128   8  34 157 198</span></span>
<span id="cb3-38"><a href="#cb3-38" tabindex="-1"></a><span class="fu">head</span>(late_early_ratios)</span>
<span id="cb3-39"><a href="#cb3-39" tabindex="-1"></a><span class="co">#&gt;       late_early_ratio</span></span>
<span id="cb3-40"><a href="#cb3-40" tabindex="-1"></a><span class="co">#&gt; 1.CEL      -0.85912360</span></span>
<span id="cb3-41"><a href="#cb3-41" tabindex="-1"></a><span class="co">#&gt; 2.CEL       0.08838817</span></span>
<span id="cb3-42"><a href="#cb3-42" tabindex="-1"></a><span class="co">#&gt; 3.CEL      -1.35966918</span></span>
<span id="cb3-43"><a href="#cb3-43" tabindex="-1"></a><span class="co">#&gt; 4.CEL      -1.14506690</span></span>
<span id="cb3-44"><a href="#cb3-44" tabindex="-1"></a><span class="co">#&gt; 5.CEL      -0.50815862</span></span>
<span id="cb3-45"><a href="#cb3-45" tabindex="-1"></a><span class="co">#&gt; 6.CEL      -0.18543936</span></span></code></pre></div>
<p>This ordering can be used directly in plotting functions or to
structure your data for further analysis, ensuring that samples are
consistently arranged by their proliferation characteristics.</p>
</div>
</div>
<div id="summarizing-subtype-metrics-with-get_subtype_metrics" class="section level2" number="1.4">
<h2><span class="header-section-number">1.4</span> 3. Summarizing
Subtype Metrics with <code>get_subtype_metrics</code></h2>
<p>Understanding how clinical or experimental variables are distributed
across molecular subtypes is a key part of downstream analysis. The
<code>get_subtype_metrics</code> function makes it easy to tabulate and
summarize metadata variables (such as gender, survival events, or other
annotations) by subtype classification.</p>
<p>This function works by matching sample IDs between your metadata and
prediction results, then counting the number of samples for each subtype
and each level of your chosen metadata variable. You can focus on a
specific value (e.g., only males, or only events) or get a full
breakdown for all levels.</p>
<div id="example-counting-events-by-subtype" class="section level3" number="1.4.1">
<h3><span class="header-section-number">1.4.1</span> Example: Counting
Events by Subtype</h3>
<p>Suppose you want to see how many progression events occurred in each
subtype:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="co"># Run the classifier if not already done</span></span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>results <span class="ot">&lt;-</span> <span class="fu">classify_samples</span>(</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a>  <span class="at">this_data =</span> sjodahl_2017,</span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a>  <span class="at">include_data =</span> <span class="cn">TRUE</span>,</span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a>  <span class="at">verbose =</span> <span class="cn">FALSE</span></span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a>)</span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a></span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a><span class="co"># Tabulate survival events (e.g., &quot;surv_os_event&quot; column)</span></span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a>metrics_events <span class="ot">&lt;-</span> <span class="fu">get_subtype_metrics</span>(</span>
<span id="cb4-10"><a href="#cb4-10" tabindex="-1"></a>  <span class="at">these_predictions =</span> results,</span>
<span id="cb4-11"><a href="#cb4-11" tabindex="-1"></a>  <span class="at">this_metadata =</span> sjodahl_2017_meta,</span>
<span id="cb4-12"><a href="#cb4-12" tabindex="-1"></a>  <span class="at">this_metadata_variable =</span> <span class="st">&quot;surv_os_event&quot;</span>,</span>
<span id="cb4-13"><a href="#cb4-13" tabindex="-1"></a>  <span class="at">factor_level =</span> <span class="dv">1</span></span>
<span id="cb4-14"><a href="#cb4-14" tabindex="-1"></a>)</span>
<span id="cb4-15"><a href="#cb4-15" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(metrics_events)</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">prediction</th>
<th align="right">total_samples</th>
<th align="right">surv_os_event_1</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">BaSq</td>
<td align="right">56</td>
<td align="right">30</td>
</tr>
<tr class="even">
<td align="left">GU</td>
<td align="right">54</td>
<td align="right">29</td>
</tr>
<tr class="odd">
<td align="left">Mes</td>
<td align="right">16</td>
<td align="right">11</td>
</tr>
<tr class="even">
<td align="left">ScNE</td>
<td align="right">20</td>
<td align="right">9</td>
</tr>
<tr class="odd">
<td align="left">Uro</td>
<td align="right">121</td>
<td align="right">58</td>
</tr>
</tbody>
</table>
</div>
<div id="example-summarizing-gender-distribution" class="section level3" number="1.4.2">
<h3><span class="header-section-number">1.4.2</span> Example:
Summarizing Gender Distribution</h3>
<p>You can also use this function to summarize categorical variables,
such as gender:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a>metrics_gender <span class="ot">&lt;-</span> <span class="fu">get_subtype_metrics</span>(</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>  <span class="at">these_predictions =</span> results,</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a>  <span class="at">this_metadata =</span> sjodahl_2017_meta,</span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a>  <span class="at">this_metadata_variable =</span> <span class="st">&quot;gender&quot;</span>,</span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a>  <span class="at">factor_level =</span> <span class="st">&quot;Male&quot;</span></span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a>)</span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(metrics_gender)</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">prediction</th>
<th align="right">total_samples</th>
<th align="right">gender_Male</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">BaSq</td>
<td align="right">56</td>
<td align="right">41</td>
</tr>
<tr class="even">
<td align="left">GU</td>
<td align="right">54</td>
<td align="right">42</td>
</tr>
<tr class="odd">
<td align="left">Mes</td>
<td align="right">16</td>
<td align="right">13</td>
</tr>
<tr class="even">
<td align="left">ScNE</td>
<td align="right">20</td>
<td align="right">14</td>
</tr>
<tr class="odd">
<td align="left">Uro</td>
<td align="right">121</td>
<td align="right">95</td>
</tr>
</tbody>
</table>
<p>This approach helps you quickly assess the distribution of key
variables across subtypes, supporting both quality control and
hypothesis generation in your analyses.</p>
</div>
</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
